packages/astro/src/integration/create-integration.ts

```diff
@@ -110,7 +110,16 @@ function createIntegration<Params extends HotloadAstroClerkIntegrationParams>()
             `
             ${command === 'dev' ? `console.log("${packageName}","Initialize Clerk: page")` : ''}
             import { runInjectionScript } from "${buildImportPath}";
-            await runInjectionScript(${JSON.stringify(internalParams)});`,
+
+            await runInjectionScript(${JSON.stringify(internalParams)});
+
+            document.addEventListener('astro:before-swap', async e => {
+              window.Clerk.__internal_flushEmotionCache();
+            });
+
+            document.addEventListener('astro:page-load', async () => {
+              await runInjectionScript(${JSON.stringify(internalParams)});
+            })`,
```

Comment:
This event only runs when view transitions is enabled. This will run on load and every page navigation.

It mounts all Clerk components in the new page.